Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -756,4 +756,20 @@ def test_analytics
     cmd("analytics", "off")
     assert_match "Analytics is disabled", cmd("analytics")
   end
+
+  def test_switch
+    assert_match "Usage: brew switch <name> <version>", cmd_fail("switch")
+    assert_match "testball not found", cmd_fail("switch", "testball", "0.1")
+
+    setup_test_formula "testball", <<-EOS.undent
+      keg_only "just because"
+    EOS
+
+    cmd("install", "testball")
+    testball_rack = HOMEBREW_CELLAR/"testball"
+    FileUtils.cp_r testball_rack/"0.1", testball_rack/"0.2"
+
+    cmd("switch", "testball", "0.2")
+    cmd_fail("switch", "testball", "0.3")
```

Comment:
I think this would also benefit from an additional `assert_match` on the command output. If I'm not mistaken, it's checking [this part](https://github.com/Homebrew/brew/blob/39453691ba3208dbd2e6ce1fd341d430b31dff55/Library/Homebrew/cmd/switch.rb#L25-L33) of the `brew switch` code, which isn't entirely obvious from the `cmd_fail` call alone.